src/shared/mount-util.c

```diff
@@ -872,12 +872,22 @@ static int mount_in_namespace(
 
         mount_tmp_created = true;
 
-        if (is_image)
-                r = verity_dissect_and_mount(FORMAT_PROC_FD_PATH(chased_src_fd), mount_tmp, options, NULL, NULL, NULL, NULL);
-        else
+        if (is_image) {
+                _cleanup_free_ char *chased_src_path = NULL;
+
+                /* For the dm-verity autodiscovery to work, we need to pass the original path. */
+                r = fd_get_path(chased_src_fd, &chased_src_path);
```

Comment:
hmm, why retrieve the path from the fd? chase_symlinks() rturns it anyway if you want? is this about pinning?

btw, the chase_symlinks() call above uses CHASE_TRAIL_SLASH flag which I figure makes no sense right now, if the path is not requested anyway...?